presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1337,10 +1337,10 @@ private void analyzeHaving(QuerySpecification node, Scope scope)
                 Expression expression = item.getSortKey();
 
                 Expression orderByExpression = null;
-                if (expression instanceof QualifiedNameReference && !((QualifiedNameReference) expression).getName().getPrefix().isPresent()) {
```

Comment:
The second part of the condition is not needed because Identifier only has one part.